scanpy/preprocessing/_simple.py

```diff
@@ -772,15 +772,26 @@ def scale_array(
             "... be careful when using `max_value` " "without `zero_center`."
         )
 
+    if np.issubdtype(X.dtype, np.integer):
+        logg.info(
+            '... scale_data: as scaling leads to float results, integer '
+            'input is cast to float, returning copy.'
+        )
+        if issparse(X):
+            X = X.__class__(X, dtype=np.float) # keep the identical sparse matrix type but with float data
+        else:
+            X = np.array(X, dtype=np.float) # keep dense array but with float data
```

Comment:
This could be a little more simple:

```suggestion
    if np.issubdtype(X.dtype, np.integer):
        logg.info(
            '... as scaling leads to float results, integer '
            'input is cast to float, returning copy.'
        )
        X = X.astype(float)
```